### DISCUSSION TITLE
Very large k values for ~200M vectors

### DISCUSSION THREAD

**Discussion by mataney** [facebookresearch/faiss#3644] [General]
Hi 👋 
Thank's for the great work!
I have a dataset of around 2M vectors of size 768 (BERT embeddings, but (a) I can lower the dimensionality, (b) for the sake of argument let's assume they are just random vectors) and I'd like to do approximate NN search on that dataset.
My requirement is that my k value is very large - around 50,000.

IIUC max k is 1024/2048, depending on my GPU.
But what if I will shard my index to 25-50 different shards? Can this do the trick?
What do you think it will do to the latency?
Is there a tutorial that shows something remotely like this?

Thank you for your help

**Comment by mdouze** [comment]
Sharding is not guaranteed to work because there is always the possibility that the 50,000 neighbors are in a single shard.
Do you need exact distance comparison?
